app/grandchallenge/evaluation/forms.py

```diff
@@ -369,6 +369,11 @@ def __init__(self, *args, user, phase: Phase, **kwargs):  # noqa: C901
     def clean_algorithm_image(self):
         algorithm_image = self.cleaned_data["algorithm_image"]
 
+        if self._phase.count_valid_archive_items == 0:
```

Comment:
The check should be on the Phase, not the algorithm image, right? The error message is also too technical, maybe something simple first like - this phase is not ready for submissions yet, then the rest.

And yes, tests would be good.

Reply:
Yes, logically this should be done in the phase clean method, but since we want to return an error to the user and we don't expose the phase field on the form, I added it to the algorithm image field. I can add the error to the non-field errors on the form, so they show up on top, if you prefer, but it's quite crowded there already with all warnings etc, so wasn't sure.

Reply:
Yes I think hoisting it is fine. Maybe this should anyway be added to all those warnings as no one will be able to submit with this new check too. It's really hard to communicate all of the tasks the admins must do to get everything set up.